Cargo.toml

```diff
@@ -67,6 +67,7 @@ convert_case = "0.6.0"
 darling = "0.20"
 data-encoding = "2"
 deno_task_shell = ">=0.13"
+derivative = "2.2.0"
```

Comment:
I think it would make it more error prone e.g. when I add a field to `ItemData` etc. then I need to remember about it. Moreover it looks like a pretty small crate https://github.com/mcarton/rust-derivative with not a lot of deps. If you insist I can implement it manually though

Reply:
It's not really about size of crate, but rather it being a proc macro :) Although, I have not benchmarked this. 

> then I need to remember about it

I believe you can avoid it by struct destructuring 

```
impl Debug for Trait {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        let Trait {
            trait_constants,
            trait_types,
            trait_functions,
            item_data,
            id: _, node: _,
        } = self;
        f.debug_struct("Trait")
            .field("trait_constants", trait_constants)
            .field("trait_types", trait_types)
            .field("trait_functions", trait_functions)
            .field("item_data", item_data)
            .finish()
    }
}
```

Reply:
Although if it changes a lot, I don't want to be overly insistive. What do you think @mkaput ?

Reply:
Okay, destructing convinces me

Reply:
Nvm, it does not - I would have to play with formatter and waste time for no reason to achieve a readable test file

Reply:
I'd use procmacro here. I'm considering using Serde here though because this data as JSON or whatever might actually by itself be a cool format for some other tooling.

Reply:
`Serde` is fine too, wdyt @maciektr?

Reply:
Sounds good 👍🏻

Reply:
https://rust-lang.github.io/rfcs/2963-rustdoc-json.html

Reply:
Done